packages/sdk/src/browser-wallets/leather/index.ts

```diff
@@ -0,0 +1,148 @@
+import { Psbt } from "bitcoinjs-lib";
+
+import {
+  BrowserWalletExtractTxFromNonFinalizedPsbtError,
+  BrowserWalletNetworkMismatchError,
+  BrowserWalletNotInstalledError,
+  getAddressFormat,
+  getNetworkByAddress,
+  OrditSDKError,
+} from "../..";
+import { BrowserWalletNetwork } from "../../config/types";
+import type { BrowserWalletSignResponse, WalletAddress } from "../types";
+import {
+  LeatherGetAddresses,
+  LeatherSignMessage,
+  LeatherSignMessageOptions,
+  LeatherSignPsbt,
+  LeatherSignPSBTOptions,
+} from "./types";
+import { leatherRequest } from "./utils";
+
+function isInstalled() {
+  if (typeof window === "undefined") {
+    throw new OrditSDKError("Cannot call this function outside a browser");
+  }
+
+  return typeof window.LeatherProvider !== "undefined";
+}
+
+async function getAddresses(
+  network: BrowserWalletNetwork = "mainnet",
+): Promise<WalletAddress[]> {
+  if (!isInstalled()) {
+    throw new BrowserWalletNotInstalledError("Leather not installed");
+  }
+
+  const res = await leatherRequest<LeatherGetAddresses>("getAddresses", {
+    network,
+  });
+
+  const addresses = res.addresses.filter(
+    (address) => address.type === "p2tr" || address.type === "p2wpkh",
+  );
+
+  // Hacky validation: there's no parameter to specify the network value when getting the address.
+  // TODO: Remove this if the wallet already supports that parameter.
+  if (getNetworkByAddress(addresses[0].address) !== network) {
+    throw new BrowserWalletNetworkMismatchError(
+      "Leather network mismatch, please switch it manually",
+    );
+  }
+
+  return addresses.map((v) => ({
+    publicKey: v.publicKey!,
+    address: v.address,
+    format: getAddressFormat(v.address, network),
+  }));
+}
+
+// Leather supports BIP-322 message signing.
+async function signMessage(
+  message: string,
+  { network = "mainnet", paymentType }: LeatherSignMessageOptions,
+): Promise<BrowserWalletSignResponse> {
+  if (!isInstalled()) {
+    throw new BrowserWalletNotInstalledError("Leather not installed");
+  }
+
+  const res = await leatherRequest<LeatherSignMessage>("signMessage", {
+    message,
+    paymentType,
+    network,
+  });
+  const { signature } = res;
```

Comment:
we can destructure directly
```suggestion
  const { signature } = await leatherRequest<LeatherSignMessage>("signMessage", {
    message,
    paymentType,
    network,
  });
```

Reply:
can